### PR TITLE
Remove pybind deps from importer and annotator

### PR DESCRIPTION
**Pull request by fortianyou**
I want to use the torch dialect importer from C++. However, the importer and annotator have inclusion usage of pybind and python. In this pull request, I refactor to remove pybind dependencies from importer and annotators' C++ sources.

Could you please review the PR @silvasean

### CODE REVIEW

File: python/torch_mlir/dialects/torch/importer/jit_ir/csrc/get_registered_ops.h
Comment (silvasean) [inline python/torch_mlir/dialects/torch/importer/jit_ir/csrc/get_registered_ops.h:19]: is this needed? I would think that "pybind.h" would already do it.

File: python/torch_mlir/dialects/torch/importer/jit_ir/csrc/pybind.cpp
Comment (silvasean) [inline python/torch_mlir/dialects/torch/importer/jit_ir/csrc/pybind.cpp:1]: rename files to class_annotator_pybind.cpp/.h

**Review by silvasean — APPROVED** [review]
Thanks so much! Other people were needing this too :)

File: python/torch_mlir/dialects/torch/importer/jit_ir/csrc/get_registered_ops.h
Comment (fortianyou) [reply python/torch_mlir/dialects/torch/importer/jit_ir/csrc/get_registered_ops.h:18]: I remove "pybind.h" since it doesn't depend on "class_annotator_pybind.h"

File: python/torch_mlir/dialects/torch/importer/jit_ir/csrc/pybind.cpp
Comment (fortianyou) [reply python/torch_mlir/dialects/torch/importer/jit_ir/csrc/pybind.cpp]: done

**Review by silvasean — APPROVED** [review]

**Comment by silvasean** [comment]
Hey Tanyo -- feel free to rebase & merge this once the CI passes (I gave you write permission). Also, can you please squash the commits before merging?

**Comment by fortianyou** [comment]
> Hey Tanyo -- feel free to rebase & merge this once the CI passes (I gave you write permission). Also, can you please squash the commits before merging?
> 
> Thanks so much!

Done, thanks!

**Comment by powderluv** [comment]
Thanks for this, multiple folks have asked for this. Maybe useful to have a simple C++ example added too.

**Pull request merged**
